wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -438,21 +437,20 @@ private FactoryDto asDto(Factory factory) throws ServerException {
     @Inject private DefaultFactoryParameterResolver defaultFactoryResolver;
 
     /**
-     * Provides a suitable resolver for the given parameters
+     * Provides a suitable resolver for the given parameters or {@code null} if none is found
+     * (generally means that provided parameters set is invalid)
      *
      * @return suitable service-specific resolver or default one
      */
     public FactoryParametersResolver getFactoryParametersResolver(Map<String, String> parameters) {
-      if (specificFactoryParametersResolvers == null) {
-        return defaultFactoryResolver;
-      }
-      for (FactoryParametersResolver factoryParametersResolver :
-          specificFactoryParametersResolvers) {
-        if (factoryParametersResolver.accept(parameters)) {
-          return factoryParametersResolver;
+      if (specificFactoryParametersResolvers != null)
```

Comment:
please don't write `if` without curly braces. It can cause nasty bugs.

Reply:
wow